src/vs/code/node/cli.ts

```diff
@@ -319,23 +324,70 @@ export async function main(argv: string[]): Promise<any> {
 			options['stdio'] = 'ignore';
 		}
 
-		const child = spawn(process.execPath, argv.slice(2), options);
+		let child: ChildProcess;
+		if (!isMacintosh) {
+			// We spawn process.execPath directly
+			child = spawn(process.execPath, argv.slice(2), options);
+		} else {
+			// On mac, we spawn using the open command to obtain behavior
+			// similar to if the app was launched from the dock
+			// https://github.com/microsoft/vscode/issues/102975
+			const openArgs: string[] = ['-n'];
+
+			if (verbose || hasReadStdinArg || args.wait) {
+				openArgs.push('-W');
+			}
 
-		if (args.wait && waitMarkerFilePath) {
-			return new Promise<void>(resolve => {
+			if (verbose) {
+				// Set up arguments to pass to open
+				const tmpStdoutName = createFileName(tmpdir(), 'code-stdout');
```

Comment:
Nit: Something tells me that a `for ... of` loop could make this code easier to read rather than building this up twice. You could have an array for `[stdout, stderr]´ and build things up from there.